src/AutoReact.ts

```diff
@@ -66,18 +88,23 @@ export default class AutoReact {
     onGreeting(message: Discord.Message) {
         let greeting = message.content.toLowerCase();
 
-        if (!greeting.startsWith("hello ") && greeting != "hello"
-            && !greeting.startsWith("hi ") && greeting != "hi"
-            && !greeting.startsWith("hey ") && greeting != "hey"
-            && !greeting.startsWith("good morning ") && greeting != "good morning"
-            && !greeting.startsWith("goodmorning ") && greeting != "goodmorning"
-            && !greeting.startsWith("good evening ") && greeting != "good evening"
-            && !greeting.startsWith("goodevening ") && greeting != "goodevening"
-            && !greeting.startsWith("good night ") && greeting != "good night"
-            && !greeting.startsWith("goodnight ") && greeting != "goodnight"
-            && !greeting.startsWith("goodday ") && greeting != "goodday"
-            && !greeting.startsWith("good day ") && greeting != "good day")
+        const words: String[] = [
+            "hello", "hi", "hey",
+            "good morning", "goodmorning",
+            "good evening", "goodevening",
+            "good night", "goodnight",
+            "good day", "goodday"
+        ];
+
+        const shouldReact = words.filter((x: string) => !greeting.startsWith(x)).filter((x: string) => greeting != x).length != words.length;
```

Comment:
You should be able to leave out the casts here if you correct the previous comments. Also, if I'm reading this correctly, it should be able to be simplified to just `words.some(x => greeting.startsWith(x))`, which looks a lot cleaner.